### PR TITLE
Split mirage into irmin-mirage, irmin-mirage-git and irmin-mirage-graphql

### LINKED ISSUE #685: split up irmin-mirage opam package into several opam packages
as suggested in #658 somewhere, I'd be in favour of splitting up the irmin-mirage opam package. I'd envision three packages at the moment:
- irmin-mirage -- the core
- irmin-mirage-graphql -- graphql adapter (depending on cohttp etc.)
- irmin-mirage-git -- depending on the git implementation

a fourth one, irmin-mirage-wodan, would be welcome as well once the irmin-wodan integration is ready.

the reasoning behind this is not only binary size, but also to minimize the dependency cone. I like my unikernels where no http code is inside ;)

the split should be fairly straightforward to do since the code is independent. please consider this for a 2.0 release. thanks!

### PR DESCRIPTION
**Pull request by zshipko**
Fixes #685

### CODE REVIEW

File: src/irmin-mirage/git/irmin_mirage_git.mli [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    (B : Irmin.Branch.S)
+    -> S
+       with type key = P.t
+        and type step = P.step
+        and module Key = P
+        and type contents = C.t
+        and type branch = B.t
+        and module Git = G
+
+  module type KV_MAKER = functor (G : Irmin_git.G) (C : Irmin.Contents.S) -> S
```
Comment (avsm) [inline src/irmin-mirage/git/irmin_mirage_git.mli]: ouch, the formatting here is just terrible. Is this via ocamlformat? This deserves a bug report if so

Comment (zshipko) [reply src/irmin-mirage/git/irmin_mirage_git.mli]: Yeah, it seems to be from ocamlformar, I will open an issue about this.

Comment (zshipko) [reply src/irmin-mirage/git/irmin_mirage_git.mli]: Actually, there appears to be an issue about this already: https://github.com/ocaml-ppx/ocamlformat/issues/671 and it looks like it's been fixed. I will investigate a little more.

File: src/irmin-mirage/git/irmin_mirage_git.ml [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+          (Irmin.Info.author info) (Irmin.Info.date info)
+          (Irmin.Info.message info)
+
+  (* XXX(samoht): extracted from Canopy. Should probably be
+       upstreamed directly. *)
+
+  module Topological = Graph.Topological.Make (S.History)
+
+  (* XXX(samoht): very slow: will go through the entire history!! *)
+  let created_updated_ids commit key =
```
Comment (samoht) [inline src/irmin-mirage/git/irmin_mirage_git.ml]: There is a new `Store.last_modified` that we should use instead.

**Comment by samoht** [comment]
Looks good, thanks! I'll create a separate issue to keep track of the use of `last_modified`

**Pull request merged**